crates/aptos/src/node/mod.rs

```diff
@@ -486,3 +495,52 @@ impl CliCommand<serde_json::Value> for ShowValidatorSet {
         Ok(response.into_inner())
     }
 }
+
+#[derive(Parser)]
+/// Update the current validator's network and fullnode addresses
+pub struct UpdateValidatorNetworkAddresses {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+
+    #[clap(flatten)]
+    pub(crate) validator_config_args: ValidatorConfigArgs,
+}
+
+#[async_trait]
+impl CliCommand<Transaction> for UpdateValidatorNetworkAddresses {
+    fn command_name(&self) -> &'static str {
+        "UpdateValidatorNetworkAddresses"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<Transaction> {
+        let (
+            consensus_public_key,
+            validator_network_public_key,
+            full_node_network_public_key,
+            validator_host,
+            full_node_host,
+        ) = self.validator_config_args.process_inputs()?;
+        let validator_network_addresses =
+            vec![validator_host.as_network_address(validator_network_public_key)?];
+        let full_node_network_addresses =
+            match (full_node_host.as_ref(), full_node_network_public_key) {
+                (Some(host), Some(public_key)) => vec![host.as_network_address(public_key)?],
+                _ => vec![],
+            };
+
+        self.txn_options
+            .submit_script_function(
+                AccountAddress::ONE,
+                "Stake",
+                "update_network_and_fullnode_addresses",
+                vec![],
+                vec![
+                    bcs::to_bytes(&consensus_public_key)?,
```

Comment:
I'm wondering, why do we need the consensus public key here?

Reply:
I don't think this should be the public_key, this is the pool address

Reply:
yeah, misunderstood the param